File: storm-spring/src/test/java/st/orm/spring/model/Visit.java
package st.orm.spring.model;

import java.time.Instant;
import java.time.LocalDate;
import org.jspecify.annotations.Nullable;
import st.orm.Entity;
import st.orm.FK;
import st.orm.PK;
import st.orm.Version;

public record Visit(
        @PK Integer id,
        LocalDate visitDate,
        @Nullable String description,
        @FK Pet pet,
        @Version Instant timestamp
) implements Entity<Integer>  {}

File: storm-kotlin-spring/src/test/kotlin/st/orm/spring/model/Visit.kt
package st.orm.spring.model

import st.orm.Entity
import st.orm.FK
import st.orm.PK
import st.orm.Version
import java.time.Instant
import java.time.LocalDate

/**
 * Simple domain object representing a visit.
 */
@JvmRecord
data class Visit(
    @PK val id: Int = 0,
    val visitDate: LocalDate,
    val description: String? = null,
    @FK val pet: Pet,
    @Version val timestamp: Instant?,
) : Entity<Int>
